packages/ai-openai/src/browser/openai-frontend-application-contribution.ts

```diff
@@ -38,72 +40,119 @@ export class OpenAiFrontendApplicationContribution implements FrontendApplicatio
-            this.manager.createOrUpdateLanguageModels(...models.map(createOpenAIModelDescription));
+            const requestSettings = this.preferenceService.get<RequestSetting[]>(PREFERENCE_NAME_REQUEST_SETTINGS, []);
+            this.manager.createOrUpdateLanguageModels(...models.map(modelId => this.createOpenAIModelDescription(modelId, requestSettings)));
             this.prevModels = [...models];
 
             const customModels = this.preferenceService.get<Partial<OpenAiModelDescription>[]>(CUSTOM_ENDPOINTS_PREF, []);
-            this.manager.createOrUpdateLanguageModels(...createCustomModelDescriptionsFromPreferences(customModels));
+            this.manager.createOrUpdateLanguageModels(...this.createCustomModelDescriptionsFromPreferences(customModels));
             this.prevCustomModels = [...customModels];
 
             this.preferenceService.onPreferenceChanged(event => {
                 if (event.preferenceName === API_KEY_PREF) {
                     this.manager.setApiKey(event.newValue);
                 } else if (event.preferenceName === MODELS_PREF) {
-                    const oldModels = new Set(this.prevModels);
-                    const newModels = new Set(event.newValue as string[]);
-
-                    const modelsToRemove = [...oldModels].filter(model => !newModels.has(model));
-                    const modelsToAdd = [...newModels].filter(model => !oldModels.has(model));
-
-                    this.manager.removeLanguageModels(...modelsToRemove.map(model => `openai/${model}`));
-                    this.manager.createOrUpdateLanguageModels(...modelsToAdd.map(createOpenAIModelDescription));
-                    this.prevModels = [...event.newValue];
+                    this.handleModelChanges(event.newValue as string[]);
                 } else if (event.preferenceName === CUSTOM_ENDPOINTS_PREF) {
-                    const oldModels = createCustomModelDescriptionsFromPreferences(this.prevCustomModels);
-                    const newModels = createCustomModelDescriptionsFromPreferences(event.newValue);
-
-                    const modelsToRemove = oldModels.filter(model => !newModels.some(newModel => newModel.id === model.id));
-                    const modelsToAddOrUpdate = newModels.filter(newModel =>
-                        !oldModels.some(model =>
-                            model.id === newModel.id &&
-                            model.model === newModel.model &&
-                            model.url === newModel.url &&
-                            model.apiKey === newModel.apiKey &&
-                            model.enableStreaming === newModel.enableStreaming));
-
-                    this.manager.removeLanguageModels(...modelsToRemove.map(model => model.id));
-                    this.manager.createOrUpdateLanguageModels(...modelsToAddOrUpdate);
+                    this.handleCustomModelChanges(event.newValue as Partial<OpenAiModelDescription>[]);
```

Comment:
Same issue with the type here. It's no longer correct as it was extended by the `defaultRequestSettings` which are not stored in this preference. We should either adapt the preference schema or the type.